### PR TITLE
[POAE7-1231] import json string of subQuery and tableColumns to ORC reader through TableLayoutHandle

### PR DESCRIPTION
**Pull request by yma11**
General Changes
import json string of subQuery and tableColumns to ORC reader through TableLayoutHandle

### CODE REVIEW

**Comment by yma11** [comment]
@winningsix @jikunshang please take a review, thanks.

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java [hunk 1]
```diff
@@ -102,8 +101,12 @@ private static JsonNode getJsonFilterNode(FilterNode filterNode)
                     operandsNode.add(operandsInput);
                     String type = hiveColumnHandle.getHiveType().toString();
                     String targetType = hiveColumnHandle.getTypeSignature().toString();
-                    operandsFeilds.put("type", type);
-                    operandsFeilds.put("target_type", targetType);
+                    operandsFeilds.put("type", matchType(type));
+                    operandsFeilds.put("target_type", matchType(targetType));
```
Comment (jikunshang) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:105]: toUpperCase, same as line 104

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java [hunk 2]
```diff
@@ -117,6 +120,25 @@ private static JsonNode getJsonFilterNode(FilterNode filterNode)
         return objectNode;
     }
 
+    private static String matchOperator(String originOperator)
+    {
+        switch (originOperator) {
+            case "EQUAL": return "=";
```
Comment (jikunshang) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:126]: match more case like GreateThan, LessThan

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java [hunk 3]
```diff
@@ -117,6 +120,25 @@ private static JsonNode getJsonFilterNode(FilterNode filterNode)
...
+        return originOperator;
+    }
+
+    private static String matchType(String type)
+    {
+        switch (type) {
+            case "int":
+            case "integer":
+            case "Int":
+                return "Integer";
```
Comment (jikunshang) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:137]: upper case, and match more type

File: presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java [hunk 4]
```diff
@@ -102,8 +101,12 @@ private static JsonNode getJsonFilterNode(FilterNode filterNode)
                     operandsNode.add(operandsInput);
                     String type = hiveColumnHandle.getHiveType().toString();
                     String targetType = hiveColumnHandle.getTypeSignature().toString();
-                    operandsFeilds.put("type", type);
-                    operandsFeilds.put("target_type", targetType);
+                    operandsFeilds.put("type", matchType(type));
+                    operandsFeilds.put("target_type", matchType(targetType));
+                    operandsFeilds.put("scale", "0");
```
Comment (jikunshang) [inline presto-hive/src/main/java/com/facebook/presto/hive/rule/HiveFilterPushdownDelegation.java:106]: these should be number rather than string

**Comment by yma11** [comment]
Hi @winningsix , you can merge this PR now. Thanks, Yan.

**Pull request merged**